src/commands/tileindex-validate/tileindex.validate.ts

```diff
@@ -522,3 +560,38 @@ export async function validate8BitsTiff(tiff: Tiff): Promise<void> {
     throw new Error(`${tiff.source.url.href} is not a 8 bits TIFF`);
   }
 }
+
+export function* iterateMapSheets(bounds: BBox, gridSize: GridSize): Generator<string> {
+  const minX = Math.min(bounds[0], bounds[2]);
+  const maxX = Math.max(bounds[0], bounds[2]);
+  const minY = Math.min(bounds[1], bounds[3]);
+  const maxY = Math.max(bounds[1], bounds[3]);
+
+  const tilesPerMapSheet = Math.floor(MapSheet.gridSizeMax / gridSize);
+
+  const tileWidth = Math.floor(MapSheet.width / tilesPerMapSheet);
+  const tileHeight = Math.floor(MapSheet.height / tilesPerMapSheet);
+
+  let tile: string = '';
+  const stepsX = calculateStep(minX, maxX, MapSheet.origin.x, tileWidth);
+  const stepsY = calculateStep(minY, maxY, MapSheet.origin.y, tileHeight);
+
+  for (let stepY = 0; stepY < stepsY; stepY += 1) {
+    const y = maxY - stepY * tileHeight;
+    for (let stepX = 0; stepX < stepsX; stepX += 1) {
+      const x = minX + stepX * tileWidth;
+      try {
+        tile = getTileName(x, y, gridSize);
+      } catch (e) {
+        if (e instanceof RangeError) {
+          logger.warn(e['message']);
+          continue;
+        } else throw e;
+      }
+      yield tile;
+    }
+  }
+  if (tile === '') {
```

Comment:
do we care if the iterator found no mapsheets? in the way we are using it, it would be cleaner to convert from a iterator into returning a full covering array.